chia/wallet/wallet_weight_proof_handler.py

```diff
@@ -56,22 +51,17 @@ async def validate_weight_proof(
         if summaries is None or sub_epoch_weight_list is None:
             raise ValueError("weight proof failed sub epoch data validation")
         validate_from = get_fork_ses_idx(old_proof, weight_proof)
-        task = asyncio.create_task(
-            validate_weight_proof_inner(
-                self._constants,
-                self._executor,
-                self._executor_shutdown_tempfile.name,
-                self._num_processes,
-                weight_proof,
-                summaries,
-                sub_epoch_weight_list,
-                skip_segment_validation,
-                validate_from,
-            )
+        valid, block_records = await validate_weight_proof_inner(
+            self._constants,
+            self._executor,
+            self._executor_shutdown_tempfile.name,
+            self._num_processes,
+            weight_proof,
+            summaries,
+            sub_epoch_weight_list,
+            skip_segment_validation,
+            validate_from,
         )
-        self._weight_proof_tasks.append(task)
-        valid, block_records = await task
-        self._weight_proof_tasks.remove(task)
```

Comment:
this transformation is only valid if there's only ever a single task that gets here. Is that a safe assumption?

Reply:
There will always only be one task in there yes because we only long sync to one peer at the time and the validation only happens as part of the long sync. But i think even if there could potentially be more tasks added i think it would be safe to drop this because we add/await/remove still or why do you think it would not be safe?